### PR TITLE
backend/scheduler-changes-for-testing-and-added-logs-for-payout

### PR DESCRIPTION
**Pull request by prashant601**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverReferee.hs [hunk 1]
```diff
@@ -73,8 +73,11 @@ linkReferee merchantId apiKey RefereeLinkInfoReq {..} = do
   unless (driverInfo.enabled) $
     throwError $ InvalidRequest "Driver is not enabled"
   let isMultipleDeviceIdExist_ = fromMaybe False isMultipleDeviceIdExist
-  unless isMultipleDeviceIdExist_ $ updateReferralStats driverInfo driverReferralLinkage.driverId
-  let flagReason = if isMultipleDeviceIdExist_ then Just DRD.MultipleDeviceIdExists else Nothing
+  merchOpCityId <- maybe (QP.findById driverReferralLinkage.driverId >>= fromMaybeM (PersonNotFound (driverReferralLinkage.driverId.getId)) <&> (.merchantOperatingCityId)) pure driverInfo.merchantOperatingCityId
+  transporterConfig <- SCTC.findByMerchantOpCityId merchOpCityId (Just (DriverId (cast driverReferralLinkage.driverId))) >>= fromMaybeM (TransporterConfigNotFound merchOpCityId.getId)
+  updateReferralStats driverReferralLinkage.driverId transporterConfig merchOpCityId
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverReferee.hs:78]: it will be updated always?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs [hunk 2]
```diff
@@ -269,8 +269,8 @@ sendReferralFCM ride booking mbRiderDetails transporterConfig = do
     fraudChecksForReferralPayout mobileNumberHash riderDetails mbDailyStats = do
       availablePersonWithNumber <- SQP.findAllMerchantIdByPhoneNo riderDetails.mobileCountryCode mobileNumberHash
       let isValidForMinPickupThreshold = maybe True (>= transporterConfig.minPickupDistanceThresholdForReferralPayout) booking.distanceToPickup
-          isValidForMinRideDistance = ride.traveledDistance >= transporterConfig.minRideDistanceThresholdForReferralPayout
-          isMaxReferralExceeded = maybe False ((< transporterConfig.maxPayoutReferralForADay) . (.activatedValidRides)) mbDailyStats
+          isValidForMinRideDistance = (fromMaybe (highPrecMetersToMeters ride.traveledDistance) ride.chargeableDistance) >= highPrecMetersToMeters transporterConfig.minRideDistanceThresholdForReferralPayout
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs:272]: We should consider ride.traveledDistance as source of truth for travel distance. Why did you change this logic?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverReferee.hs (hunk 1 above)
Comment (prashant601) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverReferee.hs:78]: yes.
